pontoon/sync/formats/l20n.py

```diff
@@ -0,0 +1,178 @@
+                    ast.String([value], value),
+                    False
+                ) for plural_form, value in self.strings.items()
+            )
+        else:
+            structure.body.remove(self.entry)
+
+
+class L20NAttributeEntity(VCSTranslation):
+    """
+    Represents an attributes of the parent entity/entry.
+    e.g.
+    <sample_entry "Sample string"
+        sample_attr: "Sample attribute">
+    Will modify only contents of sample_attr.
+    """
+    def __init__(self, entry, attr, order):
+        self.entry = entry
+        self.attr = attr
+        self.string = attr.value.source
+        super(L20NAttributeEntity, self).__init__(
+            key=".".join([entry.id.name, attr.id.name]),
+            source_string=attr.value.source,
+            strings={None: attr.value.source},
+            comments=[],
+            fuzzy=False,
+            order=order,
+        )
+
+    def __repr__(self):
+        return '<L20NAttributeEntity {key}>'.format(key=self.key.encode('utf-8'))
+
+    def update_entry(self, structure, locale):
+        if self.strings:
+            self.attr.value.content[0] = self.strings[None]
+        else:
+            self.entry.attrs.remove(self.attr)
+
+
+class L20NResource(ParsedResource):
+    def __init__(self, path, source_path, locale, structure):
+        self.path = path
+        self.source_path = source_path
+        self.locale = locale
+        self.structure = structure
+
+    @property
+    def translations(self):
+        """
+        Method maps objects from parser to the VCS translations.
+        To avoid unecessary computations, translations are memoized.
+        """
+        if not hasattr(self, '_translations'):
+            comments = []
+            translations = []
+            order = 0
+
+            for obj in self.structure.body:
+                if obj.type == 'Entity':
+                    if obj.value.type == 'String':
+                        translations.append(L20NStringEntity(obj, obj.value.source, comments, order))
+                        order += 1
+
+                    elif obj.value.type == 'Hash':
+                        strings = {self.locale.get_plural_index(item.id.name): item.value.source for item in obj.value.items}
+                        translations.append(L20NHashEntity(obj, strings, comments, order))
+                        order += 1
+
+                    for attr in obj.attrs:
+                        translations.append(L20NAttributeEntity(obj, attr, order))
+                        order += 1
+                    comments = []
+
+                elif obj.type == 'Comment':
+                    comments.append(obj.body)
+                else:
+                    comments = []
+            self._translations = translations
+        return self._translations
+
+    def save(self, locale):
+        for translation in self.translations:
+            translation.update_entry(self.structure, locale)
+
```

Comment:
You only loop parsed translations. If new translations gets added, it doesn't get saved to a file.